src/services/sessions.js

```diff
@@ -43,29 +43,86 @@ module.exports = class SessionsHelper {
 	 */
 
 	static async create(bodyData, loggedInUserId, orgId) {
-		bodyData.mentor_id = loggedInUserId
 		try {
-			const mentorDetails = await mentorExtensionQueries.getMentorExtension(loggedInUserId)
+			// If type is passed store it in upper case
+			bodyData.type && (bodyData.type = bodyData.type.toUpperCase())
+			console.log('<<<<<<<<<<<<<<bodyData.type :', bodyData.type, common.PRIVATE)
+			// If session type is private and mentorId is not passed in request body return an error
+			if (
+				bodyData.type &&
+				bodyData.type == common.PRIVATE &&
+				(!bodyData.mentor_id || bodyData.meeting_id == '')
+			) {
+				return common.failureResponse({
+					message: 'MENTORS_NOT_FOUND',
+					statusCode: httpStatusCode.bad_request,
+					responseCode: 'CLIENT_ERROR',
+				})
+			}
+
+			// session created_by and updated_by added to session creation data
+			bodyData.created_by = loggedInUserId
+			bodyData.updated_by = loggedInUserId
+			let menteeIdsToEnroll = bodyData.mentees ? bodyData.mentees : []
+			// If body data has mentor id use that id to further checks else use logged user_id
+			const mentorIdToCheck = bodyData.mentor_id || loggedInUserId
+			const isSessionCreatedByManager = !!bodyData.mentor_id
+
+			// Confirm passed id is of a mentor
+			const mentorDetails = await mentorExtensionQueries.getMentorExtension(mentorIdToCheck)
```

Comment:
@VISHNUDAS-tunerlabs session manager will not have entry in the mentor extension table,  need to change

Reply:
@rakeshSgr we are not checking mentor extension for manager. if manager is creating a session mentorIdTocheck will be mentor_id passed by manager in the req body.

Reply:
@VISHNUDAS-tunerlabs This code will execute when the session manager calls this API

Reply:
@rakeshSgr if manager is creating session mentor_id should be passed else we will throw error. and we will be always passing mentorid